src/test/ui/parser/issue-35813-postfix-after-cast.rs

```diff
@@ -0,0 +1,63 @@
+// edition:2018
```

Comment:
Please also exercise `::Try` and `::Call`, with the latter being e.g.
```rust
    type F = fn(u8);
    drop as F();
```

Reply:
Let's also exercise type ascription (`drop: F()`) for each case (+ the ones I've noted above) you've exercised for `as` here.

Reply:
Test cases added!